master/internal/root.go

```diff
@@ -0,0 +1,36 @@
+package internal
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/actor"
+)
+
+// root manages the lifecycle of all actors in the Determined master and
+// defines a supervision strategy specifically for the master.
+func root(ctx *actor.Context) error {
```

Comment:
my only concern with this is a retry loop where we spawn agents, crash and lose them over and over again. we might should land some sort of agent/task recovery first

Reply:
It may be better, rather than closing, to just print massive warnings that say to clean up resource, restart and report the error to us

Reply:
Yeah, I think I would very slightly lean toward just logging. Just keeping it a little bit less potentially disruptive as a first step. I think either way is fine, though.

Reply:
Yah, I'll start with a log.. just incase.